### PR TITLE
fix: return empty list from exceptions in fetch_columns on deltalake extractor

### PR DESCRIPTION
**Pull request by justaparth**
### Summary of Changes
For the deltalake metadata extractor, when fetching columns the code `describe`s the table, and then tries to look at schema by doing `spark.table('<table>').schema`; this is wrapped in a try-catch to fail gracefully.

We ran into an issue recently where running `spark.table('<table>').schema` raised an exception, causing this method to drop into the except and log the error and return `raw_columns`. However, the type of `raw_columns` is not `List[ScrapedColumnMetadata]`, it's a list of spark Row objects, e.g.

```
[Row(col_name='date', data_type='date', comment=''),
 Row(col_name='hour_start', data_type='timestamp', comment=None),
...
```

which caused the eventual user of the result (`create_table_metadata`) to encounter an exception because the metadata was not the type it was expecting.

This commit fixes this by returning an empty list instead of `raw_columns`, and adds a test to assert that this is the behavior if an exception is returned in `fetch_columns`.

### Tests
I added a test to simulate the behavior of `fetch_columns` with a non-existent table, which will cause an exception in the method.

### Documentation
N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [X] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [X] PR includes a summary of changes.
- [X] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: databuilder/databuilder/extractor/delta_lake_metadata_extractor.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def fetch_columns(self, schema: str, table: str) -> List[ScrapedColumnMetadata]:
                 field_dict[field.name] = field
         except AnalysisException as e:
             LOGGER.error(e)
-            return raw_columns
+            return None
```
Comment (justaparth) [inline databuilder/databuilder/extractor/delta_lake_metadata_extractor.py:302]: another option would be to return `[]` here; i thought `None` felt a bit more explicit about failure versus an empty list, but i'm open to whatever would be more consistent with the codebase broadly

**Comment by sewardgw** [comment]
Can you fix the signoff?
```
git commit --amend --no-edit --signoff
```

**Comment by justaparth** [comment]
@sewardgw done!

**Comment by feng-tao** [comment]
cc @jroof88 given you work on delta extractor recently

**Comment by jroof88** [comment]
> cc @jroof88 given you work on delta extractor recently

@justaparth and I work together 😎 and he identified this bug! PR LGTM

**Comment by justaparth** [comment]
- update to return `[]` instead of `None`; i think it should fix the CI bug and is actually closer i think to the original code's intent

**Review by feng-tao — APPROVED** [review]

**Pull request merged**